sklearn/tree/_utils.pyx

```diff
@@ -361,14 +362,8 @@ cdef class WeightedPQueue:
         # Resize if capacity not sufficient
         if array_ptr >= self.capacity:
             self.capacity *= 2
-            array = <WeightedPQueueRecord*> realloc(self.array_,
-                                                    self.capacity *
-                                                    sizeof(WeightedPQueueRecord))
-
-            if array == NULL:
-                # no free; __dealloc__ handles that
-                return -1
-            self.array_ = array
```

Comment:
It should be fine. I'm guessing it wasn't put in originally because there was no need to. If it's not using and GIL operations anyway then there shouldn't be a difference.